src/EditorFeatures/Core/EditorConfigSettings/Data/CodeStyle/CodeStyleSetting.EnumCodeStyleSetting.cs

```diff
@@ -58,6 +61,44 @@ protected override CodeStyleOption2<T> GetOption()
                 => _editorConfigOptions.TryGetEditorConfigOption(_option, out CodeStyleOption2<T>? value) && value is not null
                     ? value
                     : _visualStudioOptions.GetOption(_option);
+
+            public override string? GetSettingName()
+            {
+                // Could change to op.GetType().Name == "EditorConfigStorageLocation`1" but not sure if it applies for all editorconfig settings
+                var option = _option.StorageLocations.FirstOrDefault(op => op.GetType().Name != "RoamingProfileStorageLocation");
```

Comment:
duplicative code.  can you make this a helper somewhere tthat all these types can use instead?